<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>735</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SPH Fluid Sim</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="0">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="s_paramGB">
      <property name="title">
       <string>Simulation Parameters</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="18" column="0">
        <widget class="QDoubleSpinBox" name="m_tensionBox">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>50.000000000000000</double>
         </property>
         <property name="value">
          <double>0.072800000000000</double>
         </property>
        </widget>
       </item>
       <item row="31" column="2">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Blobby Radius</string>
         </property>
        </widget>
       </item>
       <item row="32" column="0">
        <widget class="QDoubleSpinBox" name="m_isovalue">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-2.000000000000000</double>
         </property>
         <property name="maximum">
          <double>2.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
       <item row="31" column="1">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Voxel Size</string>
         </property>
        </widget>
       </item>
       <item row="32" column="2">
        <widget class="QDoubleSpinBox" name="m_blobRadius">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>0.500000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="32" column="1">
        <widget class="QDoubleSpinBox" name="m_voxelSize">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="31" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Isovalue</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Stiffness</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Number of Particles </string>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QDoubleSpinBox" name="m_containerSizeZBox">
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>2.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QDoubleSpinBox" name="m_containerSizeYBox">
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>2.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Container Size</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QDoubleSpinBox" name="m_containerSizeXBox">
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>2.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Smoothing Radius</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QDoubleSpinBox" name="m_smoothRadiusBox">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
       <item row="22" column="0">
        <widget class="QDoubleSpinBox" name="m_gravityXBox">
         <property name="minimum">
          <double>-99.989999999999995</double>
         </property>
        </widget>
       </item>
       <item row="21" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Gravity</string>
         </property>
        </widget>
       </item>
       <item row="29" column="0">
        <widget class="QDoubleSpinBox" name="m_obstacleRadiusBox">
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>0.400000000000000</double>
         </property>
        </widget>
       </item>
       <item row="22" column="1">
        <widget class="QDoubleSpinBox" name="m_gravityYBox">
         <property name="minimum">
          <double>-99.989999999999995</double>
         </property>
         <property name="value">
          <double>-9.800000000000001</double>
         </property>
        </widget>
       </item>
       <item row="22" column="2">
        <widget class="QDoubleSpinBox" name="m_gravityZBox">
         <property name="minimum">
          <double>-99.989999999999995</double>
         </property>
        </widget>
       </item>
       <item row="17" column="1">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Viscosity</string>
         </property>
        </widget>
       </item>
       <item row="18" column="1">
        <widget class="QDoubleSpinBox" name="m_viscosityBox">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="value">
          <double>3.500000000000000</double>
         </property>
        </widget>
       </item>
       <item row="34" column="0">
        <widget class="QPushButton" name="m_runButton">
         <property name="text">
          <string>Run</string>
         </property>
        </widget>
       </item>
       <item row="34" column="1">
        <widget class="QPushButton" name="m_stopButton">
         <property name="text">
          <string>Pause</string>
         </property>
        </widget>
       </item>
       <item row="34" column="2">
        <widget class="QPushButton" name="m_resetButton">
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
       <item row="33" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0">
        <widget class="QSpinBox" name="m_numParticlesBox">
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item row="17" column="2">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Collision Elasticity</string>
         </property>
        </widget>
       </item>
       <item row="18" column="2">
        <widget class="QDoubleSpinBox" name="m_elasticityBox">
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>0.200000000000000</double>
         </property>
        </widget>
       </item>
       <item row="8" column="2">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="35" column="1">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="35" column="2">
        <widget class="QLabel" name="label_14">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>*Reset to see changes*</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
       <item row="23" column="0">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="27" column="0">
        <widget class="QDoubleSpinBox" name="m_obstacleXBox">
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
       <item row="27" column="1">
        <widget class="QDoubleSpinBox" name="m_obstacleYBox">
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
       <item row="27" column="2">
        <widget class="QDoubleSpinBox" name="m_obstacleZBox">
         <property name="minimum">
          <double>-10.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
       <item row="26" column="0" colspan="2">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Create Sphere Obstacle: Position</string>
         </property>
        </widget>
       </item>
       <item row="28" column="0">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Sphere Radius</string>
         </property>
        </widget>
       </item>
       <item row="29" column="1">
        <widget class="QPushButton" name="m_addObstacleButton">
         <property name="text">
          <string>Add Obstacle</string>
         </property>
        </widget>
       </item>
       <item row="29" column="2">
        <widget class="QPushButton" name="m_deleteObstaclesButton">
         <property name="text">
          <string>Clear Obstacles</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Rest Density</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QDoubleSpinBox" name="m_restDensityBox">
         <property name="maximum">
          <double>1500.000000000000000</double>
         </property>
         <property name="value">
          <double>50.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="17" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>SurfaceTension</string>
         </property>
        </widget>
       </item>
       <item row="7" column="2">
        <widget class="QDoubleSpinBox" name="m_gasKBox">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="value">
          <double>3.500000000000000</double>
         </property>
        </widget>
       </item>
       <item row="30" column="2">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="30" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Marching Cubes</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="36" column="0">
        <widget class="QLabel" name="label_16">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Draw Mode</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="37" column="0">
        <widget class="QComboBox" name="m_drawBox">
         <property name="currentIndex">
          <number>4</number>
         </property>
         <item>
          <property name="text">
           <string>Points</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Density</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Surface</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Velocity</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Marching Cubes</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="36" column="2">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Isosurface</string>
         </property>
        </widget>
       </item>
       <item row="37" column="2">
        <widget class="QComboBox" name="m_isosurface">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Averaged Centers</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Metaballs</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
